src/main/java/nl/dtls/fairdatapoint/service/impl/FairSearchClientImpl.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package nl.dtls.fairdatapoint.service.impl;
+
+import com.google.common.base.Preconditions;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import javax.annotation.Nonnull;
+import nl.dtls.fairdatapoint.service.FairSearchClient;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.eclipse.rdf4j.model.IRI;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service layer implementation for FairSearchClient
+ * 
+ * @author Rajaram Kaliyaperumal <rr.kaliyaperumal@gmail.com>
+ * @since 2017-10-30
+ * @version 0.1
+ */
+@Service("fairSearchClientImpl")
+public class FairSearchClientImpl implements FairSearchClient {
+    
+    private final static org.apache.logging.log4j.Logger LOGGER = LogManager.
+            getLogger(FairSearchClientImpl.class);  
+    
+    @Value("${fairSearch.fdpSubmitUrl:http://example.com/fse}")
+    private String fdpSubmitUrl;
+    
+    /**
+     * This method submit FDP url to the FAIR search engine. This method returns the Http response
+     * from the search. In case of exception Http Internal server error code (500) is returned.
+     * 
+     * @param uri   FDP base URI
+     * @return Http response code
+     */
+    @Override
+    public int submitFdpUri(@Nonnull IRI uri) {
+        Preconditions.checkState(uri != null, "FDP uri can't be null");
+        try {
+            HttpResponse<String> response = Unirest.get(fdpSubmitUrl).header("fdp", uri.toString())
+                    .asString();
+            return response.getStatus();
```

Comment:
Any specific reason to choose unirest over [httpcomponents](https://hc.apache.org/httpcomponents-client-ga/index.html)?

Reply:
For me the unirest is a bit more compact. It handles HttpClient connections and it can also map the response to the Java objects. Underneath it uses httpcomponents but take way bit of boilerplate kind of code